packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -181,9 +184,17 @@ function typeDeclaredInLib(
   if (declarationFiles.length === 0) {
     return true;
   }
-  return declarationFiles.some(declaration =>
-    program.isSourceFileDefaultLibrary(declaration),
-  );
+  return declarationFiles.some(declaration => {
+    if (program.isSourceFileDefaultLibrary(declaration)) {
+      return true;
+    }
+    const { path } = declaration;
+    return (
+      // Declared in types of runtime - Treat it as if it's from lib.
+      program.sourceFileToPackageName.get(path) === undefined &&
+      /\/node_modules\/@types\/(bun|node)\//.test(path)
+    );
+  });
```

Comment:
Waiting on more info from https://github.com/typescript-eslint/typescript-eslint/pull/9500/files#r1669456255

Reply:
Okie now that #9500 is merged, I think this is unblocked @abrahamguo? Are you waiting on us for anything else?

Reply:
Yes, looks like #9500 did not improve the issues I was targeting here.
I've opened #9608 for `URL` and I'll open another issue in the next day or so for `URLSearchParams`.

Reply:
After some investigation, it looks like `URL` and `URLSearchParams` have the same issue. It _is_ an issue, but only when `DOM` is _not_ in the `lib`. This means that in the real world most of the time, it will not be a problem — just in our integration test, which I'll fix. So #9608 shouldn't be a blocker, and I'll get this PR wrapped up!